CHANGELOG.md

```diff
@@ -1,6 +1,7 @@
 ## master / unreleased
  - [REMOVED] `chunks.NewReader` is removed as it wasn't used anywhere.
  - [REMOVED] `FromData` is considered unused so was removed.
+ - [BUGFIX] Vertical compaction was missing overlaping blocks.
```

Comment:
I think it would be apt to call it a bugfix in `selectOverlappingDirs` (applies to #559 as well if that is to be merged) because vertical compaction didn't actually miss overlapping blocks.

To explain with the same example from here https://github.com/prometheus/tsdb/pull/550#issuecomment-473490971 :
* `db.compact()` runs in a loop till `db.compactor.Plan()` returns some dirs.
* In 1st loop trip {1,2,4} would be compacted. In 2nd loop trip {1,2,4} and {3} would be picked for compaction.
* Hence eventually all overlapping blocks are compacted at the end of db.compact().